tests/unit/checkpoint/test_other_optimizer.py

```diff
@@ -0,0 +1,151 @@
+import deepspeed
+from deepspeed.ops.op_builder import FusedLambBuilder
+
+from tests.unit.common import DistributedTest
+from tests.unit.simple_model import *
+
+from tests.unit.checkpoint.common import checkpoint_correctness_verification
+
+import pytest
+
+
+class TestOtherOptimizerCheckpoint(DistributedTest):
+    world_size = 2
+
+    @pytest.mark.skipif(not deepspeed.ops.__compatible_ops__[FusedLambBuilder.NAME],
+                        reason="lamb is not compatible")
+    def test_checkpoint_unfused_optimizer(self, tmpdir):
+        config_dict = {
+            "train_batch_size": 2,
+            "steps_per_print": 1,
+            "optimizer": {
+                "type": "Lamb",
+                "params": {
+                    "lr": 0.00015
+                }
+            },
+            "gradient_clipping": 1.0,
+            "fp16": {
+                "enabled": True
+            },
+            "scheduler": {
+                "type": "OneCycle",
+                "params": {
+                    "cycle_first_step_size": 1000,
+                    "cycle_first_stair_count": 500,
+                    "cycle_second_step_size": 1000,
+                    "cycle_second_stair_count": 500,
+                    "decay_step_size": 1000,
+                    "cycle_min_lr": 0.0001,
+                    "cycle_max_lr": 0.0010,
+                    "decay_lr_rate": 0.001,
+                    "cycle_min_mom": 0.85,
+                    "cycle_max_mom": 0.99,
+                    "decay_mom_rate": 0.0
+                }
+            }
+        }
+
+        args = args_from_dict(tmpdir, config_dict)
+        hidden_dim = 10
+
+        models = [SimpleModel(hidden_dim, empty_grad=False) for _ in range(2)]
+
+        def _test_checkpoint_unfused_optimizer(args,
+                                               models,
+                                               hidden_dim,
+                                               load_optimizer_states):
+            checkpoint_correctness_verification(
+                args,
+                models=models,
+                hidden_dim=hidden_dim,
+                tmpdir=tmpdir,
+                load_optimizer_states=load_optimizer_states)
+
+        _test_checkpoint_unfused_optimizer(args=args,
+                                           models=models,
+                                           hidden_dim=hidden_dim,
+                                           load_optimizer_states=True)
+
+        _test_checkpoint_unfused_optimizer(args=args,
+                                           models=models,
+                                           hidden_dim=hidden_dim,
+                                           load_optimizer_states=False)
```

Comment:
```suggestion
        checkpoint_correctness_verification(
            args,
            models=models,
            hidden_dim=hidden_dim,
            tmpdir=tmpdir,
            load_optimizer_states=True)
            
        checkpoint_correctness_verification(
            args,
            models=models,
            hidden_dim=hidden_dim,
            tmpdir=tmpdir,
            load_optimizer_states=False)
```

This might be easier to read (and is less code) than using a local function

Reply:
We could even make this a for loop:
`for load_opt_state in [True, False]:`

Reply:
Done